x-pack/plugins/apm/public/context/apm_service/apm_service_context.tsx

```diff
@@ -63,6 +67,16 @@ export function ApmServiceContextProvider({
     agentName,
   });
 
+  // Replace transactionType in the URL in case it is not one of the types returned by the API
+  useMemo(() => {
+    if (
```

Comment:
Does it make sense to centralise the logic introduced by https://github.com/elastic/kibana/pull/110336 as well?

Reply:
Maybe that is no longer needed, I'm going to test it.

Reply:
Thanks, there are more `replace` calls in `public/components/app/transaction_details/index.tsx` and `public/components/app/transaction_overview/index.tsx`

Reply:
It'd be great if we can wrap all this in e.g. a <RedirectToServiceTransactionType/> component.

Reply:
@cauemarcondes could you please check transaction_details and transaction overview as per above?

Reply:
@cauemarcondes just to double check this in case it was missed